From 83de4be01b8b8585e275f3fb05692bc886c64441 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Wed, 5 Jun 2024 16:10:00 +0200
Subject: [PATCH 0900/1695] plat-stm32mp1: conf: disable compilation of PKA and
 SAES on a/d chips.

Do not compile PKA and SAES drivers on
STM32MP1xa/d as the IP not present.
Do not compile PKA and SAES drivers on
configuration that does not have RNG IP in OP-TEE.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8c3173099b88068092f2f924e43e7ec7e3b27ee1 plat-stm32mp1: conf: disable compilation of PKA and SAES on a/d chips.]
Change-Id: I1bafacceb1a12be92b08aac92c4df4aa9028fc85
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/385513
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 52cfa5c90..76b55b9f0 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -170,11 +170,14 @@ endif
 
 ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-no_cryp)),)
 $(call force,CFG_STM32_CRYP,n)
+$(call force,CFG_STM32_PKA,n)
 $(call force,CFG_STM32_SAES,n)
 endif
 
 ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-no_rng)),)
 $(call force,CFG_HWRNG_PTA,n)
+$(call force,CFG_STM32_PKA,n)
+$(call force,CFG_STM32_SAES,n)
 CFG_WITH_SOFTWARE_PRNG ?= y
 endif
 
-- 
2.47.2

